compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2647,6 +2648,13 @@ public void visit(BLangInvocation invocationExpr) {
         }
     }
 
+    public void visit(BLangErrorConstructorExpr errorConstructorExpr) {
+        analyzeExprs(errorConstructorExpr.positionalArgs);
+        if (errorConstructorExpr.namedArgs.isEmpty()) {
```

Comment:
Shouldn't this be

```suggestion
        if (!errorConstructorExpr.namedArgs.isEmpty()) {
```

Please check if we can add a test for this too.

Reply:
Added